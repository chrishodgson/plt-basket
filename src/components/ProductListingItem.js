import React from 'react';
import { create } from "react-test-renderer";
import { productListingItem } from '../../src/components/ProductListingItem'

export default ( {product: {img, name, price}} ) => {
    return (
        <div className="productListingItem">
            <img src={img} alt={name} />
            <div className="productListingItemText">
                <div>{name}</div>
                <div>&pound;{price.toFixed(2)}</div>
            </div>
        </div>
    );
};
