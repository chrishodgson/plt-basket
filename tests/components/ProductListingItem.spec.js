import React from "react";
import { create } from "react-test-renderer";
import ProductListingItem from '../../src/components/ProductListingItem'

describe("ProductListingItem component", () => {
    test("ProductListingItem matches the snapshot", () => {
        const product = { img: 'image url', name: 'trousers', price: 9 };
        const component = create(<ProductListingItem product={product} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});