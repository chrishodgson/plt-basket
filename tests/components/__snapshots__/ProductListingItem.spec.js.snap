// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductListingItem component ProductListingItem matches the snapshot 1`] = `
<div
  className="productListingItem"
>
  <img
    alt="trousers"
    src="image url"
  />
  <div
    className="productListingItemText"
  >
    <div>
      trousers
    </div>
    <div>
      Â£
      9.00
    </div>
  </div>
</div>
`;
